juriscraper/opinions/united_states/state/sd.py

```diff
@@ -3,106 +3,59 @@
-            if not neutral_cite:
-                neutral_cite = self.extract_regex_group(1, s, True)
-            # Make the citation SD instead of S.D. The former is a neutral cite, the latter, the South Dakota Reporter
-            neutral_cites.append(neutral_cite.replace(".", "").upper())
-        return neutral_cites
 
-    def _download_backwards(self, page_year):
-        logger.info(f"Running with params: {page_year}")
-        self.initiate_webdriven_session()
-
-        # Select the year (this won't trigger a GET unless it's changed)
-        path = "//*[@id='ContentPlaceHolder1_PageContent_OpinionYears']/option[@value={year}]".format(
-            year=page_year[1]
+    def _process_html(self):
+        rows = self.html.xpath(
+            "//div[@id='ContentPlaceHolder1_ChildContent1_UpdatePanel_Opinions']//tbody/tr"
         )
-        option = self.find_element_by_xpath(path)
-        option.click()
-
-        if page_year[0] != 0:
-            # Not the first, page, go to the one desired.
-            links = self.find_elements_by_xpath(
-                "//a[@href[contains(., 'Page')]]"
+        for row in rows:
+            cite = re.findall(r"\d{4} S\.D\. \d+", get_row_column_text(row, 2))
+            if not cite:
+                continue
+
+            # https://ujs.sd.gov/uploads/sc/opinions/2928369ef9a6.pdf
+            # We abstract out the first part of the docket number here
+            # And process the full docket number in the `extract_from_text` method
+            # Called after the file has been downloaded.
+            docket = row.xpath(".//td[2]/a/@href")[0].split("/")[-1][:5]
+            self.cases.append(
+                {
+                    "date": get_row_column_text(row, 1),
+                    "name": titlecase(
+                        get_row_column_text(row, 2).rsplit(",", 1)[0]
+                    ),
+                    "neutral_citation": cite[0],
+                    "url": row.xpath(".//td[2]/a/@href")[0],
+                    "docket": docket,
+                }
             )
-            links[page_year[0] - 1].click()
-
-        text = self._clean_text(self.webdriver.page_source)
-        self.webdriver.quit()
-        html_tree = html.fromstring(text)
-
-        html_tree.rewrite_links(
-            fix_links_in_lxml_tree, base_href=self.request["url"]
-        )
-        self.html = html_tree
-        self.status = 200
 
-    def extract_regex_group(self, group, string, alt=False):
-        regex = self.regex_alt if alt else self.regex
-        return re.search(regex, string, re.MULTILINE).group(group)
+    def extract_from_text(self, scraped_text: str) -> Dict[str, Any]:
+        """Can we extract the date filed from the text?
+
+        :param scraped_text: The content of the document downloaded
+        :return: Metadata to be added to the case
+        """
+        docket = re.findall(r"#\d+.*-.-\w{3}", scraped_text)[0][:20]
```

Comment:
Sorry, I keep mentioning this, but when I see `[:20]`, I worry. Can you put a comment about what's going on here with an example of what's getting parsed?

Reply:
sounds good